### PR TITLE
Do not fail user deletion when a remote notification fails

### PR DESCRIPTION
**Pull request by smatting**
## Checklist

https://wearezeta.atlassian.net/browse/FS-124

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information:
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

**Review by pcapriotti — APPROVED** [review]
Looks good to me.

File: services/brig/src/Brig/IO/Intra.hs [hunk 1]
```diff
@@ -278,9 +278,6 @@ notifyUserDeletionRemotes deleted = do
           case eitherFErr of
             Left fErr -> do
               logFederationError (tDomain uids) fErr
-              -- FUTUTREWORK: Do something better here?
-              -- FUTUREWORK: Write test that this happens
-              throwM $ federationErrorToWai fErr
             Right () -> pure ()
```
Comment (mdimjasevic) [inline services/brig/src/Brig/IO/Intra.hs:281]: A pattern like this is encapsulated by `whenLeft :: Applicative m => Either a b -> (a -> m ()) -> m ()`.

File: services/brig/test/integration/API/User/Account.hs [hunk 2]
```diff
@@ -1257,6 +1258,56 @@ testDeleteWithRemotes opts brig = do
...
+          then F.OutwardResponseError (F.OutwardError F.ConnectionRefused "mocked connection problem with b domain")
+          else OutwardResponseBody (cs $ Aeson.encode EmptyResponse)
+
+  let galleyHandler :: ReceivedRequest -> MockT IO Wai.Response
+      galleyHandler (ReceivedRequest requestMethod requestPath _requestBody) =
+        case (requestMethod, requestPath) of
+          (_methodDelete, ["i", "user"]) -> do
+            let response = Wai.responseLBS Http.status200 [(Http.hContentType, "application/json")] (cs $ Aeson.encode EmptyResponse)
+            pure response
+          _ -> error "not mocked"
```
Comment (mdimjasevic) [inline services/brig/test/integration/API/User/Account.hs:1288]: There's no need to pattern match on `requestMethod` as it is vacuous, right?

File: services/galley/test/integration/API.hs [hunk 3]
```diff
@@ -3239,23 +3248,34 @@ removeUser = do
...
+          | otherwise = error "unmocked domain"
+
     (_, fedRequests) <-
-      withTempServantMockFederator (const emptyFederatedBrig) galleyApi localDomain $
+      withTempMockFederator' handler $
         deleteUser alexDel' !!! const 200 === statusCode
 
     liftIO $ do
-      assertEqual ("expect exactly 5 federated requests in : " <> show fedRequests) 5 (length fedRequests)
+      assertEqual ("expect exactly 5 federated requests in : " <> show fedRequests) 7 (length fedRequests)
```
Comment (mdimjasevic) [inline services/galley/test/integration/API.hs:3278]: Should the quoted string also have 7 in it instead of 5?

**Review by mdimjasevic — APPROVED** [review]
This looks good. A few smaller comments are left inlined.

File: services/brig/src/Brig/IO/Intra.hs (hunk 1 above)
Comment (smatting) [reply services/brig/src/Brig/IO/Intra.hs]: Ah nice, done.

File: services/brig/test/integration/API/User/Account.hs (hunk 2 above)
Comment (smatting) [reply services/brig/test/integration/API/User/Account.hs]: Yes, you're right. I changed this to a proper pattern match in https://github.com/wireapp/wire-server/pull/1912/commits/d92e7fa976992f97f17733d0147c2fecd0d7e88b

**Pull request merged**